src/test/java/seedu/address/logic/commands/FindCommandTest.java

```diff
@@ -154,6 +156,34 @@ public void execute_nameAndModuleRoleKeywords_multiplePersonsFound() throws Pars
         assertEquals(Arrays.asList(CARL, ELLE), model.getFilteredPersonList());
     }
 
+    @Test
+    public void execute_nameAndModuleRoleAndTags_onePersonFound() throws ParseException {
+        String expectedMessage = String.format(MESSAGE_PERSONS_LISTED_OVERVIEW, 1,
+            "(Benson) AND (CS1101S-Tutor) AND (friends)");
+        NameContainsKeywordsPredicate namePredicate = prepareNamePredicate("Benson");
+        ModuleRoleContainsKeywordsPredicate moduleRolePredicate = prepareModuleRolePredicate("CS1101S-TA");
+        TagContainsKeywordsPredicate tagPredicate = prepareTagPredicate("friends");
+        List<Predicate<Person>> predicates = Arrays.asList(namePredicate, moduleRolePredicate, tagPredicate);
+
+        FindCommand command = new FindCommand(predicates);
+        expectedModel.updateFilteredPersonList(namePredicate.and(moduleRolePredicate).and(tagPredicate));
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Arrays.asList(BENSON), model.getFilteredPersonList());
+    }
+
+    @Test
+    public void execute_nonExistentTag_noPersonFound() throws ParseException {
+        String expectedMessage = String.format(MESSAGE_PERSONS_LISTED_OVERVIEW, 0,
+            "(invalid)");
+        TagContainsKeywordsPredicate tagPredicate = prepareTagPredicate("invalid");
```

Comment:
small nit but i think it might be better to use a different tag value to avoid confusion? since this is a valid tag

Reply:
Hmmm true, my brain screw up at this moment haha, I think replace it with `nothing` should be fine